src/request.ts

```diff
@@ -231,6 +236,20 @@ class Request {
     const cloudUrl: string = Auth.getEndpointForResource(hostname, cloudType);
     options.url = options.url!.replace(hostname, cloudUrl);
   }
+
+  private createProxyConfigFromString(url: string): AxiosProxyConfig {
```

Comment:
Since we're expecting the string to be a URL, let's rename the method to `createProxyConfigFromUrl` to make it clearer. Otherwise, it'll be confusing down the road to know what the method is meant to do without looking at its implementation.